tests/eosio.system_tests.cpp

```diff
@@ -1998,6 +1998,30 @@ BOOST_FIXTURE_TEST_CASE(voter_gbm_pay, eosio_system_tester, * boost::unit_test::
    }
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(burning_preminted, eosio_system_tester, * boost::unit_test::tolerance(1e-10)) try {
+   cross_15_percent_threshold();
+
+   create_account_with_resources(N(user11111111), config::system_account_name, core_sym::from_string("100.0000"), false,
+                                 core_sym::from_string("10.0000"), core_sym::from_string("10.0000"));
+
+   // This transer creates a sub_balance for genesis.wax account
+   transfer( N(eosio), N(genesis.wax), core_sym::from_string("100000000.0000"), N(eosio) );
+
+   uint64_t nonce = 1;
+   asset genesis_tokens_user1{core_sym::from_string("50000000.0000")};
+   awardgenesis(N(user11111111), genesis_tokens_user1, nonce);
+
+   produce_block( fc::days(1) ); //Wait until July 1st
+   produce_block( fc::days(1096 / 2) );
+
+   BOOST_REQUIRE_EQUAL(success(), unstake( N(user11111111), N(user11111111), core_sym::from_string("25000000.0000"), core_sym::from_string("25000000.0000")));
+   produce_blocks(1);
+
+   asset balance = get_balance(N(genesis.wax));
+   // Failing: critical check balance == core_sym::from_string("50000000.0000") has failed [49999981.4879 TST != 50000000.0000 TST]
```

Comment:
Could be the case, that the seen difference comes from **user11111111** unstaking (undelegatebw) being the one paying for the table modifications here ?

Reply:
The point is that genesis.wax account has less then the expected balance. I'm not checking the user11111111 balance.